packages/rx_bloc_test/README.md

```diff
@@ -40,4 +42,51 @@ rxBlocTest<DetailsBloc,String>(
   skip: 3, // This will skip first two values + the initial value
   expect: ['Hello world'],
 )
+```  
+
+## rxBlocFakeAsyncTest
+For the most complex test cases with multiple events, which contain `throttleTime` or/and `debounceTime` or similar inside, and to speed up test execution,`rxBlocFakeAsyncTest` should be used. It contains an instance of [FakeAsync](https://pub.dev/packages/fake_async) inside `act`, which provides a way to fire all asynchronous events that are scheduled for that time period without actually needing the test to wait for real time to elapse.
+
+**Example**
+
+[DetailsBloc](https://github.com/Prime-Holding/rx_bloc/blob/develop/packages/rx_bloc_test/test/helpers/details_bloc/details_bloc.dart) email state mapper: 
+```dart
+@override
+Stream<String> _mapToEmailState() => _$setEmailEvent
+  .startWith('')
+  .throttleTime(const Duration(seconds: 3), trailing: true)
+  .map((email) => email.trim())
+  .shareReplay(maxSize: 1);
+```
+
+✅ **GOOD - execution time ±35 ms:**
+```dart
+rxBlocFakeAsyncTest<DetailsBloc, String>(
+  'Email fake async test bloc',
+  build: () => DetailsBloc(repo),
+  state: (bloc) => bloc.states.email,
+  act: (bloc, fakeAsync) {
+    bloc.events.setEmail(' job');
+    bloc.events.setEmail(' job@prime');
+    bloc.events.setEmail(' job@prime.com ');
+    fakeAsync.elapse(const Duration(seconds: 3));
+  },
+  expect: <String>['', 'job@prime.com'],
+);
+```
+
+
+❌ **BAD - execution time ±3 sec:**
+```dart
+rxBlocTest<DetailsBloc, String>(
+  'Email test bloc',
+  build: () async => DetailsBloc(repo),
+  state: (bloc) => bloc.states.email,
+  act: (bloc) async {
+    bloc.events.setEmail(' job');
+    bloc.events.setEmail(' job@prime');
+    bloc.events.setEmail(' job@prime.com ');
+  },
```

Comment:
I guess here we need to add await Future.delayed so we can highlight what the difference will be?

Reply:
No, example contain real-word scenario with throttleTime, please pay attention to Email state mapper on line 52.

Reply:
![](https://user-images.githubusercontent.com/5969391/170605597-e0431337-e685-441e-9df2-650bfeafaa2c.png)